### PR TITLE
feat: add upsert all support

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Added support for a new upsertAll api. This can be used for upserting multiple configs in fewer calls. Eventually, we'll optimize its internals to also reduce the roundtrips to the persistence store, but this change encapsulates that tech debt.

### Testing
Added and updated existing UTs

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java [hunk 1]
```diff
@@ -23,34 +26,42 @@
...
+
+  /**
+   * Write each of the provided config value associated with the specified config resource to the
+   * store.
+   *
+   * @param resourceContextValueMap
+   * @param userId
+   * @return the upserted configs
+   */
+  List<UpsertedConfig> writeAllConfigs(
```
Comment (surajpuvvada) [inline config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java:63]: what is the error handling behavior ?

File: config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java [hunk 2]
```diff
@@ -64,68 +80,69 @@ private Datastore initDataStore(Config config) {
...
     }
   }
 
   @Override
-  public ContextSpecificConfig getConfig(ConfigResource configResource) throws IOException {
-    return getConfig(configResource, getLatestVersion(configResource));
+  public List<UpsertedConfig> writeAllConfigs(
+      Map<ConfigResourceContext, Value> resourceContextValueMap, String userId) throws IOException {
+    // TODO keep pushing this down into doc store
+    List<UpsertedConfig> list = new ArrayList<>(resourceContextValueMap.size());
```
Comment (surajpuvvada) [inline config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java:126]: so would docstore be able to leverage bulk upsert APIs ?

Comment (aaron-steinfeld) [reply config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java:126]: That's the idea - just split the PR. The current PR handles all the API changes, which is a fairly significant refactor, but doesn't really change the implementation in any meaningful way. A future PR needs only swap out this implementation to use the doc store's bulk upsert.

File: config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java (hunk 1 above)
Comment (aaron-steinfeld) [reply config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java:63]: Currently, it's identical to calling N write operations sequentially. If a later one fails, it aborts at that point leaving the original (this is the same behavior in the current abstraction one layer up in ObjectStore). Ideally, with a doc store API change we can make this transactional with all or nothing semantics (this came up with Entity Service in conversations with @saxenakshitiz too, but I believe as of today doc store itself doesn't support this - another reason I paused this change where I did).

**Pull request merged**